packages/gatsby-theme-newrelic/src/hooks/useInstrumentedHandler.js

```diff
@@ -0,0 +1,35 @@
+import { useCallback, useEffect, useRef } from 'react';
+import warning from 'warning';
+
+const useInstrumentedHandler = (handler, attributes) => {
+  const savedHandler = useRef();
+
+  useEffect(() => {
+    savedHandler.current = handler;
+  }, [handler]);
+
+  return useCallback(
+    (...args) => {
+      const { actionName, ...attrs } =
+        typeof attributes === 'function' ? attributes(...args) : attributes;
+
+      if (window.newrelic) {
+        warning(
+          actionName,
+          'You are attempting to instrument a handler, but the `actionName` property is not set. This will result in a no-op.'
+        );
+
+        if (actionName) {
+          actionName && window.newrelic.addPageAction(actionName, attrs);
```

Comment:
Is the `actionName` necessary if we're already in the `if` block?

Reply:
Oops, I could have sworn I removed the `if`, but must have hit `u` by accident 😅 . Will remove the `if`

Reply:
Fixed in a752daf